### ISSUE TITLE
Problems in Moodle 3.5

### ISSUE THREAD

**Issue by germanvaleroelizondo** [frankkoch/moodle-mod_studentquiz#263]
Moodle (3.5+)
With debuggin set to DEVELOPER:
    
line 1583 of /lib/dml/moodle_database.php: call to debugging()
    line 1543 of /lib/dml/moodle_database.php: call to moodle_database->get_record_sql()
    line 1522 of /lib/dml/moodle_database.php: call to moodle_database->get_record_select()
    line 1263 of /lib/questionlib.php: call to moodle_database->get_record()
    line 1317 of /lib/questionlib.php: call to question_get_top_category()
    line 487 of /question/editlib.php: call to question_make_default_categories()
    line 328 of /question/editlib.php: call to question_build_edit_resources()
    line 134 of /mod/studentquiz/viewlib.php: call to question_edit_setup()
    line 118 of /mod/studentquiz/viewlib.php: call to mod_studentquiz_view->load_questionbank()
    line 82 of /mod/studentquiz/view.php: call to mod_studentquiz_view->__construct()

**Comment by germanvaleroelizondo** [comment]
I noticed this issue. It disappeared when i removed StudentQuiz, but re-appeared when I re-installed it again and tried to  use it.

**Comment by timhunt** [comment]
You have given the stack trace from the error report, but not the actual error message. We really need the error message too.

**Comment by germanvaleroelizondo** [comment]
![imagen](https://user-images.githubusercontent.com/2671923/92727786-4bdfdd00-f335-11ea-9ae0-365ba5f06d21.png)

**Comment by germanvaleroelizondo** [comment]
When I click on 'Continue':
line 1543 of /lib/dml/moodle_database.php: call to moodle_database->get_record_sql()
line 1522 of /lib/dml/moodle_database.php: call to moodle_database->get_record_select()
line 1263 of /lib/questionlib.php: call to moodle_database->get_record()
line 1317 of /lib/questionlib.php: call to question_get_top_category()
line 487 of /question/editlib.php: call to question_make_default_categories()
line 328 of /question/editlib.php: call to question_build_edit_resources()
line 134 of /mod/studentquiz/viewlib.php: call to question_edit_setup()
line 118 of /mod/studentquiz/viewlib.php: call to mod_studentquiz_view->load_questionbank()
line 82 of /mod/studentquiz/view.php: call to mod_studentquiz_view->__construct()

**Comment by timhunt** [comment]
None of that is the actual error message!

**Comment by germanvaleroelizondo** [comment]
"_Salida de error, por lo que se deshabilita redireccionamiento automático._" translates to

"Error output, so disabling automatic redirect."

This string is **erroroutput | core_error**

**Comment by timhunt** [comment]
Yes. i worked that out. As you can see, it is not telling ou the error we need to know. It is say that because some error occured (which you still have not told us) the normal redirect that woudl happen is not happening. We want to see the acutal error that tells us what the problem is.

**Comment by dionysius** [comment]
A redirect should happen without output. Since you enabled Developer logging you get developer output. and output is output. developing debugging is only for developers.

**Comment by dionysius** [comment]
Essentially there is no effective error. It makes no sense to give output when a redirect happens. You should not use developer logging when using your moodle productive.

**Comment by germanvaleroelizondo** [comment]
I apologize for the inconvenience caused. My fault..

Yes. I have set debugging to NORMAL, and no error is shown. Issue solved.

I had  set debugging to DEVELOPER because I was installing several additional plugins in my experimental server (no students currently using it much).

I arrived here because the **stack trace display showed whenever I used the question bank**, e.g, when making a standard MC question, and I had no clue as to the cause. Searching in Moodle forums showed several forum threads, and one was about StudentQuiz. Removing StudentQuiz removed the stack trace display. And thus is why I arrived here.

Thanks for the support. Have a nice day.
